### ISSUE TITLE
Revision 2 of the gateway

### ISSUE THREAD

**Issue by Dexruus** [banksy-git/lidl-gateway-freedom#33]
Less of an Issue and more of an "advice"; Revision 2 of the Lidl gateway (rounded edges) is the "exact same". Its firmware just has ssh disabled. You need to TTL onto the device and start its dropbear ssh server with;

dropbear -p "port"

From thereon you are able to ssh into the device with your given root until next restart.

Its recommended to follow the tutorial part in which you set the ssh port to "always on" and "port 22":

if [ ! -f /tuya/ssh_monitor.original.sh ]; then cp /tuya/ssh_monitor.sh /tuya/ssh_monitor.original.sh; fi

echo "#!/bin/sh" >/tuya/ssh_monitor.sh

ALSO FOR NOOBS LIKE ME;

IF:

Unable to negotiate with "ip.address" "port" : no matching host key type found. Their offer: ssh-rsa,ssh-dss

USE:

ssh -o HostKeyAlgorithms=+ssh-rsa "ip-address" -p "port"

**Comment by bmix** [comment]
Would you mind uploading a photo of those parts of the device where one could draw conclusions as to how to open that darn thing?  How did you get it to open?
Thanks!

I have the rev 2, but all instructions help is for rev 1.

**Comment by GAS85** [comment]
> Would you mind uploading a photo of those parts of the device where one could draw conclusions as to how to open that darn thing? How did you get it to open? Thanks!
> 
> I have the rev 2, but all instructions help is for rev 1.

Have had the same issue. Problem was in USB connection of USB TTY device. Pull it out and in - problem was solved.

**Comment by GAS85** [comment]
> Its firmware just has ssh disabled.

Not really, it is simply running on port `2333`, this is not standard port of ssh. So, to connect you have to run command (there is no space in `-p2333`):

```shell
ssh -o HostKeyAlgorithms=+ssh-rsa root@<ip-address> -p2333
```

**Comment by sbonaime** [comment]
> Less of an Issue and more of an "advice"; Revision 2 of the Lidl gateway (rounded edges) is the "exact same". Its firmware just has ssh disabled. You need to TTL onto the device and start its dropbear ssh server with;
> 
> dropbear -p "port"
> 
> From thereon you are able to ssh into the device with your given root until next restart.
> 
> Its recommended to follow the tutorial part in which you set the ssh port to "always on" and "port 22":
> 
> if [ ! -f /tuya/ssh_monitor.original.sh ]; then cp /tuya/ssh_monitor.sh /tuya/ssh_monitor.original.sh; fi
> 
> echo "#!/bin/sh" >/tuya/ssh_monitor.sh
> 
> ALSO FOR NOOBS LIKE ME;
> 
> IF:
> 
> Unable to negotiate with "ip.address" "port" : no matching host key type found. Their offer: ssh-rsa,ssh-dss
> 
> USE:
> 
> ssh -o HostKeyAlgorithms=+ssh-rsa "ip-address" -p "port"

Hi,
How do you TTL to the device ? Is it exactly the same as V1 ? I only receive bunch of data when I connect through serial.  When do I have to hit "Esc" ? Just once ? 
Thanks

**Comment by sbonaime** [comment]
My USB/Serial is I think 5v... and not 3.3v. 

<img width="878" height="294" alt="Image" src="https://github.com/user-attachments/assets/47631edd-588d-4874-8368-ad1b714484be" />

I will try with raspberry Pi uart which is 3.3V

**Comment by Dexruus** [comment]
Yes, exact same as V1. Wellp, if you connected 5V on the 3.3V the possibility is high, that you just let go of the magic smoke inside of the machine, eg fried your ttl gate.

I ALWAYS use the powersupply of the circuitboad (here the 5V USB) to supply the board with power and never only the chip if i can.

And yes, just spam ESC until you see the prompt "login as root"

**Comment by sbonaime** [comment]
I connected to the serial port using the Pi UART  :-)
